chat-commands.js

```diff
@@ -887,6 +894,72 @@ exports.commands = {
 		}
 	},
 
+	subroom: function (target, room, user) {
+		if (!this.can('makeroom')) return;
+		if (!target) return this.parse('/help subroom');
+
+		if (!room.chatRoomData) return this.errorReply(`Temporary rooms cannot be subrooms.`);
+		if (!room.isPrivate) return this.errorReply(`Public rooms cannot be subrooms.`);
+		if (room.mainRoom) return this.errorReply(`This room is already a subroom. To change which room this subroom belongs to, remove the subroom first.`);
+
+		let main = Rooms(toId(target));
+
+		if (!main) return this.errorReply(`The room '${target}' does not exist.`);
+		if (main.isPrivate || !main.chatRoomData) return this.errorReply(`Only public rooms can have subrooms.`);
+
+		room.mainRoom = main.id;
+		main.subRooms.set(room.id, room);
+
+		let mainIdx = Rooms.global.chatRoomData.findIndex(r => r.title === main.title);
+		let subIdx = Rooms.global.chatRoomData.findIndex(r => r.title === room.title);
+
+		// This is needed to ensure that the main room gets loaded before the subroom.
+		if (mainIdx > subIdx) {
+			let tmp = Rooms.global.chatRoomData[mainIdx];
+			Rooms.global.chatRoomData[mainIdx] = Rooms.global.chatRoomData[subIdx];
+			Rooms.global.chatRoomData[subIdx] = tmp;
+		}
+
+		room.chatRoomData.mainRoom = main.id;
+		Rooms.global.writeChatRoomData();
+
+		return this.addModCommand(`This room was set as a subroom of ${main.title}.`);
+	},
+
+	unsubroom: function (target, room, user) {
+		if (!this.can('makeroom')) return;
+		if (!room.mainRoom) return this.errorReply(`This room is not currently a subroom of a public room.`);
+
+		let main = Rooms(this.mainRoom);
+
+		if (main) {
+			main.subRooms.delete(room.id);
+		}
+
+		room.mainRoom = null;
+
+		delete room.chatRoomData.mainRoom;
```

Comment:
what about combining these to

```Javascript
room.chatRoomData.mainRoom = delete room.mainRoom;
Rooms.global.writeChatRoomData();
```
It seems weird that you're setting ``room.mainRoom`` to null but ``room.chatRoomData.mainRoom`` to ``undefined``

Reply:
Your suggestion simply doesn't work.

Reply:
Deleting the entry in `chatRoomData` is important for making sure it's not stored long-term in the chatroom data JSON file. Setting the room's own `room.mainRoom` to null is ideal for performance. So leaving the code as currently written is my recommendation, too.